sim/battle-stream.ts

```diff
@@ -209,33 +209,32 @@ export function getPlayerStreams(stream: BattleStream) {
 			},
 		}),
 	};
+	const SPLIT_REGEX =  /\n\|split\|([^\n]*)\n([^\n]*)\n([^\n]*)/g;
 	(async () => {
 		let chunk;
 		// tslint:disable-next-line:no-conditional-assignment
 		while ((chunk = await stream.read())) {
 			const [type, data] = splitFirst(chunk, `\n`);
 			switch (type) {
-			case 'update':
-				const p1Update = data.replace(/\n\|split\n[^\n]*\n([^\n]*)\n[^\n]*\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-				streams.p1.push(p1Update);
-				const p2Update = data.replace(/\n\|split\n[^\n]*\n[^\n]*\n([^\n]*)\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-				streams.p2.push(p2Update);
-				// p3 and p4 share update information with p1 and p2 respectively.
-				streams.p3.push(p1Update);
-				streams.p4.push(p2Update);
-				const specUpdate = data.replace(/\n\|split\n([^\n]*)\n[^\n]*\n[^\n]*\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-				streams.spectator.push(specUpdate);
-				const omniUpdate = data.replace(/\n\|split\n[^\n]*\n[^\n]*\n[^\n]*/g, '');
-				streams.omniscient.push(omniUpdate);
+			case 'update': {
```

Comment:
What’s with the braces inside a switch statement? Isn’t that inconsistent with the style of the repository?

Reply:
Its required in certain in ES6 to create a block scope to get around the issue declaring variables with `let` or `const` which could conflict with other `case` statements. We do it elsewhere in the repo as well.

In this case though, I think the collision no longer exists, so I can remove them.